### ISSUE TITLE
[Axios Error: Request failed with error code 500]

### ISSUE THREAD

**Issue by Untuouchum** [ninoseki/mihari#525]
### **Rule doesn't generate alerts**

Hello. First thanks for your software !
I installed it (see versions used below) and added a rule which is parsed correctly. But when I run it I receive an error in webui: 
"Axios Error: Request failed with error code 500"

### **Steps to reproduce**

- Create a rule for example Evilginx2:
title: Evilginx2
description: A rule to find EvilGinx2 default service JARM
queries:
- analyzer: censys
  query: services.jarm.fingerprint:20d14d20d21d20d20c20d14d20d20daddf8a68a1444c74b6dbe09910a511e6
allowed_data_types:
- ip

**and run it.**

### **Expected behavior**
Rule generate alerts.

### **Actual behavior**
"Axios Error: Request failed with error code 500"

in logs:
(...)
/var/lib/gems/2.7.0/gems/grape-1.6.2/lib/grape/router.rb:72:in `identity'
/var/lib/gems/2.7.0/gems/grape-1.6.2/lib/grape/router.rb:56:in `block in call'
/var/lib/gems/2.7.0/gems/grape-1.6.2/lib/grape/router.rb:136:in `with_optimization'
/var/lib/gems/2.7.0/gems/grape-1.6.2/lib/grape/router.rb:55:in `call'
/var/lib/gems/2.7.0/gems/grape-1.6.2/lib/grape/api/instance.rb:165:in `call'
/var/lib/gems/2.7.0/gems/grape-1.6.2/lib/grape/api/instance.rb:70:in `call!'
/var/lib/gems/2.7.0/gems/grape-1.6.2/lib/grape/api/instance.rb:65:in `call'
/var/lib/gems/2.7.0/gems/grape-1.6.2/lib/grape/api.rb:81:in `call'
/var/lib/gems/2.7.0/gems/mihari-4.11.0/lib/mihari/web/app.rb:65:in `call'
/var/lib/gems/2.7.0/gems/mihari-4.11.0/lib/mihari/web/middleware/error_notification_adapter.rb:12:in `block in call'
/var/lib/gems/2.7.0/gems/mihari-4.11.0/lib/mihari/mixins/error_notification.rb:12:in `with_error_notification'
/var/lib/gems/2.7.0/gems/mihari-4.11.0/lib/mihari/web/middleware/error_notification_adapter.rb:11:in `call'
/var/lib/gems/2.7.0/gems/mihari-4.11.0/lib/mihari/web/middleware/connection_adapter.rb:12:in `block in call'
/var/lib/gems/2.7.0/gems/mihari-4.11.0/lib/mihari/mixins/database.rb:8:in `with_db_connection'
/var/lib/gems/2.7.0/gems/mihari-4.11.0/lib/mihari/web/middleware/connection_adapter.rb:11:in `call'
/var/lib/gems/2.7.0/gems/rack-cors-1.1.1/lib/rack/cors.rb:100:in `call'
/var/lib/gems/2.7.0/gems/rack-2.2.4/lib/rack/common_logger.rb:38:in `call'
/var/lib/gems/2.7.0/gems/puma-6.0.0/lib/puma/configuration.rb:268:in `call'
/var/lib/gems/2.7.0/gems/puma-6.0.0/lib/puma/request.rb:89:in `block in handle_request'
/var/lib/gems/2.7.0/gems/puma-6.0.0/lib/puma/thread_pool.rb:343:in `with_force_shutdown'
/var/lib/gems/2.7.0/gems/puma-6.0.0/lib/puma/request.rb:88:in `handle_request'
/var/lib/gems/2.7.0/gems/puma-6.0.0/lib/puma/server.rb:430:in `process_client'
/var/lib/gems/2.7.0/gems/puma-6.0.0/lib/puma/server.rb:233:in `block in run'
/var/lib/gems/2.7.0/gems/puma-6.0.0/lib/puma/thread_pool.rb:150:in `block in spawn_thread'
2023-01-29 22:09:15 +0100 Rack app ("GET /api/rules/87a0ff4b-bb57-31c4-8825-2fe23e7dcf1d/run" - (::1)): #<NoMethodError: undefined method `[]' for nil:NilClass>

### **Screenshots**
<img width="1681" alt="Screenshot 2023-01-29 at 21 51 22" src="https://user-images.githubusercontent.com/32979004/215355916-67e77a62-63fc-41ba-b893-3c71c2f7a010.png">

### **System Information:**

- OS: [Ubuntu 20.04.5 LTS]

- Ruby version: [Installed from apt: 
ruby --version
ruby 2.7.0p0 (2019-12-25 revision 647ee6f091) [x86_64-linux-gnu]]

- Mihari version: [e.g. 2.0.0]

**Comment by ninoseki** [comment]
It looks like that the rule hits with many artifacts (= IP addresses).
It causes a timeout issue in Puma. I will make it possible to use more lengthy timeout. 

As a workaround, you can use `search` command. It does not have timeout issue.
```bash
mihari search your_rule.yaml
```

**Comment by ninoseki** [comment]
I released a new version (v4.12.0) which allows to set `worker_timeout` value to run a time-consuming search.
For example, 
```bash
mihari web -worker-timeout 360
```
will be enough for the rule you shared.
